sockeye/training.py

```diff
@@ -747,11 +757,16 @@ def _update_metrics(self,
                               "time-elapsed": time.time() - self.state.start_tic}
         gpu_memory_usage = utils.get_gpu_memory_usage(self.model.context)
         checkpoint_metrics['used-gpu-memory'] = sum(v[0] for v in gpu_memory_usage.values())
+        checkpoint_metrics['converged'] = self.state.converged
+        checkpoint_metrics['diverged'] = self.state.diverged
 
+        # copy values from checkpoint_metrics, skipping NaNs and infinities that break numpy's histograms
         for name, value in metric_train.get_name_value():
-            checkpoint_metrics["%s-train" % name] = value
+            if np.isfinite(value):
+                checkpoint_metrics["%s-train" % name] = value
         for name, value in metric_val.get_name_value():
-            checkpoint_metrics["%s-val" % name] = value
+            if np.isfinite(value):
```

Comment:
actually what are we guarding against here? I tried invoking an issue by setting value to `np.inf`, but that didn't cause any issues. Was this related to Tensorboard? If so I think we should remove values only for tensorboard logging (or for whichever part can not handle `inf`). For the metric file itself I would argue that having the `inf` is actually valuable for debugging, no?

Reply:
So this seems to be caused by `add_histogram` in `TensorboardLogger`. So to guard against this problem we should add the check (+ warning) there. The check for finite for scalars we can then keep.

Reply:
Moving the check to TensorboardLogger.log_metrics() as per offline discussion.